src/sentry/search/events/types.py

```diff
@@ -85,6 +85,8 @@ class SnubaParams:
     start: datetime | None = None
     end: datetime | None = None
     stats_period: str | None = None
+    # granularity is used with timeseries requests to specifiy bucket size
+    granularity_secs: int = 0
```

Comment:
Thoughts on 
```suggestion
    granularity_secs: int | None = None
```
so the check for granularity is `self.granularity_secs is not None` which is a bit more explicit?

Reply:
I would prefer it like this and i do fine it weird that for table queries `granularity` is still set. But this was put into place to get around annoying typing situations. Otherwise any function that expects `granularity_secs` to be defined has to make this check (like `validate_granularity` or `TimeSeriesRequest`).

Another option I think would be add `@property def timeseries_granularity(self) -> int` which performs this check for us, this would be similar to `snubaParams.start_date` vs `snubaParams.start`.

Reply:
> which performs this check for us, this would be similar to snubaParams.start_date vs snubaParams.start.

IMO Let's continue this pattern

Reply:
👍 Fair, better to be consistent